examples/MomentoRedisLambdaExample/MomentoRedisExampleLambdaHandler/src/MomentoRedisExampleLambdaHandler/MomentoCache.cs

```diff
@@ -0,0 +1,188 @@
+    {
+        var withCompressionStopWatch = new Stopwatch();
+        Console.WriteLine($"Beginning get request for {key}");
+        var stopWatch = Stopwatch.StartNew();
+        if (compressValue)
+        {
+            withCompressionStopWatch = Stopwatch.StartNew();
+        }
+        var getResponseTask = _client.GetAsync(_cacheName, key);
+        Console.WriteLine($"\tIssued get request for {key}");
+        var getResponse = await getResponseTask;
+        Console.WriteLine($"\tDone awaiting get response for {key}");
+        
+        if (withSleep)
+        {
+            await Task.Delay(500);
+            Console.WriteLine($"GET returned from sleeping 500ms for key {key}");
+        }
+        
+        stopWatch.Stop();
+        var responseElapsedMillis = stopWatch.ElapsedMilliseconds;
+        long withDecompressionElapsedMillis = 0;
+        long responseSize = 0;
+        long decompressedResponseSize = 0;
+        if (getResponse is CacheGetResponse.Hit hitResponse)
+        {
+            Console.WriteLine($"\tResponse for {key} is a hit; decompressing");
+            responseSize = hitResponse.ValueByteArray.Length;
+            if (compressValue)
+            {
+                var decompressed = Compression.Decompress(hitResponse.ValueByteArray);
+                Console.WriteLine($"\tResponse finished decompressing for {key}");
+                decompressedResponseSize = decompressed.Length;
+                withCompressionStopWatch.Stop();
+                withDecompressionElapsedMillis = withCompressionStopWatch.ElapsedMilliseconds;
+            }
+            
+        }
+        else
+        {
+            Console.WriteLine(
+                $"\n\nGOT SOMETHING BESIDES A HIT FOR KEY {key}; PROBABLY SHOULDN'T USE THIS DATA\n\n");
+        }
+        
+        var getResult = new GetResult(
+            Response: getResponse,
+            ResponseSize: responseSize,
+            DecompressedResponseSize: decompressedResponseSize,
+            ElapsedMillis: responseElapsedMillis,
+            ElapsedMillisWithDecompression: withDecompressionElapsedMillis
+        );
+        Console.WriteLine($"\tReturning get response for {key}: {getResult}");
+        return getResult;
+    }
+    
+    public async void LogMomentoSetResult(List<Task<SetResult>> momentoSetTasks)
+    {
+        Console.WriteLine("------- MOMENTO SET RESULT -------");
+        foreach(var setTask in momentoSetTasks)
+        {
+            var setResult = await setTask;
+            if (setResult.Response is CacheSetResponse.Error err) {
+                Console.WriteLine("ERROR: " + err.Message);
+            } else if (setResult.Response is CacheSetResponse.Success) {
+                Console.WriteLine("Set in " + setResult.ElapsedMillis);
+            }
+        }
+    }
+    
+    public async void LogMomentoGetResult(List<Task<GetResult>> momentoGetTasks, bool compressValue)
```

Comment:
same comment here, better to realize this list of tasks first if possible and make this function synchronous.